pymanopt/core/problem.py

```diff
@@ -73,6 +73,7 @@ def precon(x, d):
         self._backends = list(
             filter(lambda b: b.is_available(), [
```

Comment:
One needs to test for `PytorchBackend`'s availability before `AutogradBackend`'s because both check that the objective is callable and only the former tests the argument.

Reply:
It is clear that using `arg=torch.Tensor()` to indicate that one wants the pytorch backend is a hack. To make things dirtier, one side effect of this hack is to provide a container for the cache. There are surely cleaner ways to do all this, at the expense of more work as usual. In the end, this is your project and you must decide what makes sense for you.